skyportal/models.py

```diff
@@ -101,23 +104,37 @@ class Group(Base):
 
 class Stream(Base):
     name = sa.Column(sa.String, unique=True, nullable=False)
-    url = sa.Column(sa.String, unique=True, nullable=False)
-    username = sa.Column(sa.String)
-    password = sa.Column(sa.String)
+    altdata = sa.Column(JSONB, nullable=True,
+                        doc="Misc. metadata stored in JSON format, e.g. "
+                            "`{'collection': 'ZTF_alerts', selector: [1, 2]}`")
 
-    groups = relationship('Group', secondary='stream_groups',
+    groups = relationship('Group', secondary='group_streams',
                           back_populates='streams',
                           passive_deletes=True)
+    users = relationship('User', secondary='stream_users',
+                         back_populates='streams',
+                         passive_deletes=True)
+    filters = relationship('Filter',
```

Comment:
Thanks for bringing this up! It made me think about the ACL stuff in `api/filters.py` and I rewrote it so that now all the necessary checks are in place (I think -- needs review; updated the tests accordingly).

I strongly prefer the current scheme and believe that switching to the proposed "simplified" scheme would actually complicate things in practice, and is _a lot_ of work. If you assign streams to filters then the user who does it must have the corresponding SU permissions (remember, this is a very "privileged" task), and this will have to be done for every user filter, and we will have potentially multiple filters per program. Whereas in the current scheme, as long as you are a group admin, you can create as many filters as you like on the streams the group has access to, without asking SUs for permissions -- I think it is much more user-friendly.

I also don't quite see what needs to be "manually kept in sync". It is forbidden to change both `group_id` and `stream_id` for a filter; deleting a stream from a group deletes corresponding filters as well, see `test_post_new_filter_delete_stream_deletes_filter` in `test_groups.py`.